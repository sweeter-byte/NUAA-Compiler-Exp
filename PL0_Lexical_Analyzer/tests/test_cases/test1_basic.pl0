{test1_basic.pl0 - Basic Functionality Test}
{Test: Basic syntax structure, keywords, identifiers, constants}

program test1;
const max := 100;
var x, y, z;
begin
    x := 10;
    y := 20;
    z := x + y
end