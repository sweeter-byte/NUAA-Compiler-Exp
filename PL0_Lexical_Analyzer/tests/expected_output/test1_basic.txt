================================================================================
PL/0 Lexical Analysis Results
================================================================================

Index  Line   Column Type            Value               
--------------------------------------------------------------------------------
1      4      1      PROGRAM         program             
2      4      9      ID              test1               
3      4      14     SEMICOLON       ;                   
4      5      1      CONST           const               
5      5      7      ID              max                 
6      5      11     ASSIGN          :=                  
7      5      14     INT             100                 
8      5      17     SEMICOLON       ;                   
9      6      1      VAR             var                 
10     6      5      ID              x                   
11     6      6      COMMA           ,                   
12     6      8      ID              y                   
13     6      9      COMMA           ,                   
14     6      11     ID              z                   
15     6      12     SEMICOLON       ;                   
16     7      1      BEGIN           begin               
17     8      5      ID              x                   
18     8      7      ASSIGN          :=                  
19     8      10     INT             10                  
20     8      12     SEMICOLON       ;                   
21     9      5      ID              y                   
22     9      7      ASSIGN          :=                  
23     9      10     INT             20                  
24     9      12     SEMICOLON       ;                   
25     10     5      ID              z                   
26     10     7      ASSIGN          :=                  
27     10     10     ID              x                   
28     10     12     PLUS            +                   
29     10     14     ID              y                   
30     11     1      END             end                 

================================================================================

Lexical analysis completed, no errors.

================================================================================
